app/assets/javascripts/components/option-select.js

```diff
@@ -107,14 +115,18 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
       }
     }
 
-    obj.$allCheckboxes.hide()
-    for (var j = 0; j < showCheckboxes.length; j++) {
-      obj.$allCheckboxes.eq(showCheckboxes[j]).show()
+    for (var j = 0; j < obj.$allCheckboxes.length; j++) {
+      obj.$allCheckboxes[j].style.display = 'none'
+    }
+
+    for (var k = 0; k < showCheckboxes.length; k++) {
+      obj.$allCheckboxes[showCheckboxes[k]].style.display = 'block'
     }
 
-    var len = showCheckboxes.length || 0
-    var lenChecked = obj.$optionsContainer.find('.govuk-checkboxes__input:checked').length
-    obj.$filterCount.html(len + (len === 1 ? obj.filterTextSingle : obj.filterTextMultiple) + ', ' + lenChecked + obj.filterTextSelected)
+    var lenChecked = obj.$optionsContainer.querySelectorAll('.govuk-checkboxes__input:checked').length
+    var len = (showCheckboxes.length || 0) + lenChecked
```

Comment:
```suggestion
    var len = showCheckboxes.length + lenChecked
```

I'm not sure there's a situation where length needs to be coerced to 0 - I think it's always an array.

Reply:
Probably just a literal translate.  Updated to remove the coercion.